clients/web/src/views/layout/HeaderUserView.js

```diff
@@ -13,10 +13,9 @@ girder.views.LayoutHeaderUserView = girder.View.extend({
         },
 
         'click a.g-logout': function () {
-            girder.restRequest({
-                path: 'user/authentication',
-                type: 'DELETE'
-            }).done(_.bind(function () {
+            girder.logout();
+
+            girder.events.once('g:logout.success', _.bind(function () {
```

Comment:
I'm not completely sure what you mean, but I think you're saying that there's no reason to set ``girder.currentUser`` here specifically - it can just be a general thing that Girder does by listening to the login/logout events.  Is that right?